spec/unit/hanami/cli/commands/app/generate/action_spec.rb

```diff
@@ -1,299 +1,460 @@
 # frozen_string_literal: true
 
+require "hanami"
 require "hanami/cli/commands/app/generate/action"
 require "ostruct"
 
-RSpec.describe Hanami::CLI::Commands::App::Generate::Action do
+RSpec.describe Hanami::CLI::Commands::App::Generate::Action, :app do
   subject { described_class.new(fs: fs, inflector: inflector, generator: generator) }
 
   let(:fs) { Dry::Files.new(memory: true) }
   let(:inflector) { Dry::Inflector.new }
   let(:generator) { Hanami::CLI::Generators::App::Action.new(fs: fs, inflector: inflector) }
-  let(:app) { "Bookshelf" }
-  let(:slice) { "main" }
+  let(:app) { Hanami.app.namespace }
+  let(:dir) { inflector.underscore(app) }
   let(:controller) { "users" }
   let(:action) { "index" }
   let(:action_name) { "#{controller}.#{action}" }
 
-  before { prepare_slice! }
+  context "generate for app" do
+    it "generates action" do
+      within_application_directory do
+        subject.call(name: action_name)
 
-  it "generates action" do
-    subject.call(slice: slice, name: action_name)
+        # Route
+        routes = <<~CODE
+          # frozen_string_literal: true
 
-    # route
-    expect(fs.read("config/routes.rb")).to match(%(get "/users", to: "users.index"))
+          require "hanami/routes"
 
-    # action
-    expect(fs.directory?(directory = "slices/#{slice}/actions/#{controller}")).to be(true)
+          module #{app}
+            class Routes < Hanami::Routes
+              define do
+                root { "Hello from Hanami" }
+                get "/users", to: "users.index"
```

Comment:
Just out of interest, is the leading slash in `"/users"` here actually necessary? I might be remembering things wrong, but I recall mostly not using it...

Reply:
@timriley IDK if that would work, maybe accidentally.

But why would you declare a route w/o the leading slash?
That is a Relative URL, which has the leading slash.

Reply:
I must just be mis-remembering, all good!